src/main/java/seedu/duke/commands/SetBudgetCommand.java

```diff
@@ -12,14 +16,24 @@
             + System.lineSeparator() + "|| Parameters: SET /b[BUDGET]"
             + System.lineSeparator() + "|| Example: SET /b 100.00" + System.lineSeparator();
 
+    /**
+     * Constructs the set budget command with the user-specified budget.
+     *
+     * @param amount Budget specified by the user.
+     */
     public SetBudgetCommand(double amount) {
         super();
         this.amount = amount;
+        LOGGER.log(Level.INFO,"(Set budget) User entered budget: " + amount);
     }
 
     @Override
     public CommandResult execute() {
         double correctBudget = myBudget.setBudget(amount);
+        LOGGER.log(Level.INFO,"(Set budget) Budget was corrected to: " + correctBudget);
+        assert correctBudget <= 5000 : "Budget should not exceed $5000";
+        assert correctBudget >= 0 : "Budget should not be negative";
+        LOGGER.log(Level.INFO,"(Set budget) Budget amount: " + correctBudget + " did not fail assertions.");
```

Comment:
Not sure if logging the passing of assertions is needed, since we can tell if the assertions passed if the program continues to run

Reply:
Hmm I'll just keep it here for now, not very sure how logging will interact with assertions.